From 4fc2441dd084c858ce6173f6c08d98131e57b528 Mon Sep 17 00:00:00 2001
From: Anil Dsouza <anil.dsouza@einfochips.com>
Date: Fri, 11 Feb 2022 10:59:08 +0530
Subject: [PATCH 1/4] media: i2c: ap1320: Performance improvement for ar0430
 sensor

Following changes are done with this commit
1. MEDIA BUS FORMAT is changed from UYVY8_2x8 to RGB565_2X8_LE
2. FPS is increased from 3 to 10

Change-Id: I44db095f40aef936ce1c2c5f44c04d25809bb02c
Signed-off-by: Anil Dsouza <anil.dsouza@einfochips.com>
---
 drivers/media/i2c/ap1302.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/media/i2c/ap1302.c b/drivers/media/i2c/ap1302.c
index 0a5bac91e..f63f2b514 100644
--- a/drivers/media/i2c/ap1302.c
+++ b/drivers/media/i2c/ap1302.c
@@ -482,9 +482,9 @@ struct ap1302_firmware_header {
 
 static const struct ap1302_format_info supported_video_formats[] = {
 	{
-		.code = MEDIA_BUS_FMT_UYVY8_2X8,
-		.out_fmt = AP1302_PREVIEW_OUT_FMT_FT_YUV_JFIF
-			 | AP1302_PREVIEW_OUT_FMT_FST_YUV_422,
+		.code = MEDIA_BUS_FMT_RGB565_2X8_LE,
+		.out_fmt = AP1302_PREVIEW_OUT_FMT_FT_RGB
+			 | AP1302_PREVIEW_OUT_FMT_FST_RGB_565,
 	},
 };
 
@@ -535,8 +535,8 @@ static const struct ap1302_sensor_info ap1302_sensor_info[] = {
 		.name = "ar0430",
 		.i2c_addr = 0x36,
 		.resolution = { 2316, 1746},
-		.fps = 3,
-		.format = MEDIA_BUS_FMT_UYVY8_2X8,
+		.fps = 10,
+		.format = MEDIA_BUS_FMT_RGB565_2X8_LE,
 		.supplies = (const struct ap1302_sensor_supply[]) {
 			{ "vddpll", 0 },
 			{ "vaa", 0 },
-- 
2.17.1

